server/daemon/src/opt.rs

```diff
@@ -76,6 +76,12 @@ struct HealthCheckArgs {
     /// Disable TLS verification
     #[clap(short, long, action)]
     no_verify_tls: bool,
+    /// Specify the CA cert file path to use for TLS verification
+    #[clap(short='C', long,env="KANIDM_CA_CERT")]
+    ca_cert: Option<String>,
+    /// Check the 'origin' URL from the server configuration file, instead of the 'address'
+    #[clap(short='O', long, action)]
+    check_origin: bool,
```

Comment:
If orgin goes via a load balancer this could be a huge footgun because if something happens then every servers healthcheck will fail and it'll spiral. It could also mean instances arle healthchecking partner instances not themself.

Reply:
But that depends on if one *intends* to go via the origin or not - the default is to use the `address` field, but in my use case I could use it to identify if something in the path was down/funky, including the cert having expired.

As of the current update I just pushed, if the user just supplies the config file, it'll go to the `https://address:port` and turn off TLS validation (since the bind address is rarely the hostname in a cert). If one wants to do a full end-to-end check on a system then they can healthcheck with extra options.

Reply:
But this isn't intended to be a full end to end check, it's just meant to be for "is my container still running"

I think we should limit it to local host only IMO. For end to end, we should have a different approach than this.

Reply:
I think some of these ideas are good but bigger than this issue perhaps especially the cert validation etc.

Reply:
My original intent for it was an "is this both **responding** _and_ **healthy**" which I'd expect to include:

- cert validation, especially since we're requiring TLS end-to-end and certs have a habit of expiring.
- (eventually) the introspection to be able to deal with "replication is good", the system's queue is not overloaded etc.

The origin check is kinda handy for prechecks in dev etc too, making sure the cert's valid.

Reply:
Also I think that maybe we are looking at this as different things. You're looking at healthcheck as a "lets check the whole kanidm fleet" I'm thinking about it as a local "is this one instance okay". That could be part of the thing here about the origin thing.

Reply:
Yeah, I can see both sides, hopefully folks are smart enough to know the difference - and it's ~5 lines difference between the two, with a default for the "container" check 😄